resources/resources.go

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft and contributors.  All rights reserved.
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+package resources
+
+import (
+	"context"
+	"net/http"
+	"net/url"
+
+	"github.com/Azure-Samples/azure-sdk-for-go-samples/helpers"
+	"github.com/Azure-Samples/azure-sdk-for-go-samples/iam"
+	"github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2017-05-10/resources"
+	"github.com/Azure/go-autorest/autorest"
+)
+
+func getResourcesClient() resources.Client {
+	token, _ := iam.GetResourceManagementToken(iam.AuthGrantType())
+	resourcesClient := resources.NewClient(helpers.SubscriptionID())
+	resourcesClient.Authorizer = autorest.NewBearerAuthorizer(token)
+	resourcesClient.AddToUserAgent(helpers.UserAgent())
+	return resourcesClient
+}
+
+// GetResource gets a resource, the generic way.
+// The API version parameter overrides the API version in
+// the SDK, this is needed because not all resources are
+// supported on all API versions.
+func GetResource(ctx context.Context, provider, resourceType, resourceName, apiVersion string) (resources.GenericResource, error) {
```

Comment:
This is nice. Can we rename `provider` to `resourceProvider` to match the next two parameter names? Thanks!

Reply:
Fixed